### PR TITLE
Update Hello World and add REPL example

### PR DESCRIPTION
**Pull request by tsandall**
These are the changes that went into the latest version of the Hello World example up on openpolicyagent.org. Have a look at the example and let me know you think it's good for the initial release.

[EDIT: Also, if you have any feedback on the REPL example that would be great.]

http://www.openpolicyagent.org/docs/examples/hello-world.html

### CODE REVIEW

File: docs/docs/examples/authz.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+                "Authz-User": "bob"
+            }
+        }
+        EOF
+
+    Currently, Docker does not provide a way to authenticate clients. In Docker 1.12, clients can be authenticated using TLS
+    and there are plans to include other means of authentication. For the purpose of this example, we assume that an authentication
+    system is place.
+
+1. Insert user data into OPA:
```
Comment (timothyhinrichs) [inline docs/docs/examples/authz.md]: Could we explain that we're adding this data, which could either come from an external system or as part of policy.

File: docs/docs/examples/authz.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+            --data-urlencode \
+            'global=request:{"Body":{"HostConfig":{"SecurityOpt":["seccomp=default"]}}}'
+
+    Congratulations! You have successfully prevented containers from running without seccomp!
+
+    So far, the policy has been defined in terms of input data from the plugin. In many cases, it's necessary to write policies against multiple data sources.
+
+    The rest of the example shows how you can grant fine grained access to specific clients. To do so, we will insert fake user data into OPA to simulate an authentication system.
+
+1. Create a Docker configuration file to include an extra HTTP header in all requests:
```
Comment (timothyhinrichs) [inline docs/docs/examples/authz.md]: Could we add a note in the text saying that we're faking the fact that all requests are coming from bob?

File: docs/docs/examples/authz.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        allow_request = true :- valid_user_role
+
+        # valid_user_role defines a document that is the boolean value true if this is
+        # a write request and the user is allowed to perform writes.
+        valid_user_role :-
+            req.Headers["Authz-User"] = user_id,
+            users[user_id] = user,
+            req.Method != "GET",
+            user.readOnly = false
+
```
Comment (timothyhinrichs) [inline docs/docs/examples/authz.md]: I feel like we can simplify this logic a bit.

**Comment by timothyhinrichs** [comment]
Few nits, but otherwise looks good to me.

File: docs/docs/examples/repl.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+Steps
+=====
+
+1. First, make sure you can run the REPL on your machine:
+
+        opa run
+
+    Without any data, you can experiment with simple boolean expressions to get the hang of it:
+
```
Comment (timothyhinrichs) [inline docs/docs/examples/repl.md]: I feel we could build this up a bit more--break it into smaller pieces.  This REPL isn't like anything anyone has seen before.  The explanation below is good.  But I feel some more intro material would be helpful.

File: docs/docs/examples/repl.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        +---+
+        | X |
+        +---+
+        | 0 |
+        | 1 |
+        | 2 |
+        | 3 |
+        +---+
+
+1. The REPL can also take policies and data as input. Let's define a bit of JSON data that will be used in the example:
```
Comment (timothyhinrichs) [inline docs/docs/examples/repl.md]: Might be worth saying that you need to quit the REPL, run these commands from the command line, and restart OPA

**Comment by timothyhinrichs** [comment]
Another nit or two, but looks good.

File: docs/docs/examples/repl.md (hunk 4 above)
Comment (tsandall) [inline docs/docs/examples/repl.md]: Agreed. Will come back to this after the comprehension work is done.

**Pull request merged**